### PR TITLE
Synthetic data updates

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Updated `Usage.md` with a new method description.
- Extended `test_examples` with the example.

### Reason for changes

- On top of #2979.

### Related tickets

- 152550

### Tests

- test_examples/531/ - success
- test_examples_3.10/117/ - success

### CODE REVIEW

File: tests/cross_fw/examples/run_example.py [hunk 1]
```diff
@@ -176,6 +176,14 @@ def llm_tune_params() -> Dict[str, float]:
...
 
 
+def llm_compression_synthetic() -> Dict[str, float]:
+    from examples.llm_compression.openvino.tiny_llama_synthetic_data.main import main as llm_compression_synthetic_main
+
+    result = llm_compression_synthetic_main()
+
+    return {"wikitext_based_similarity": result[0], "synthetic_based_similarity": result[1]}
+
+
```
Comment (KodiaqQ) [inline tests/cross_fw/examples/run_example.py:186]: This example run would increase the overall time for the test_examples scope significantly. Only this example would take ~30 minutes to run on the local machine.
@alexsu52, are you sure that we should add this into the scope?

Comment (KodiaqQ) [reply tests/cross_fw/examples/run_example.py:186]: More precise example timings, @alexsu52, @MaximProshin 
image
~30 minutes take the example in the test environment including package installation & etc.

Comment (KodiaqQ) [reply tests/cross_fw/examples/run_example.py:186]: The reduction of the subset & dataset size to 64, reduces the time from ~20 mins to ~16 mins for the example
image

File: examples/llm_compression/openvino/tiny_llama_synthetic_data/main.py [hunk 2]
```diff
@@ -81,28 +57,42 @@ def main():
...
+    hf_model.model = optimized_model
+    hf_model.request = None
+    opt_output = hf_model.generate(**input_ids, max_new_tokens=max_new_tokens)
+    opt_output_text = tokenizer.decode(opt_output[0])
 
-    synthetic_based_similarity = validate_model(evaluator, hf_model, synthetic_optimized_model, original_ov_model)
-    print(f"Synthetic-quantized model similarity: {synthetic_based_similarity}")
-    return wikitext_based_similarity, synthetic_based_similarity
+    print(f"Optimized model output: {opt_output_text}\n")
+    return opt_output_text
```
Comment (KodiaqQ) [inline examples/llm_compression/openvino/tiny_llama_synthetic_data/main.py:95]: For how it takes ~10 minutes for the example (without env installation).

File: docs/usage/post_training_compression/weights_compression/Usage.md [hunk 3]
```diff
@@ -61,6 +61,15 @@ nncf_dataset = nncf.Dataset(data_source, transform_fn)
 compressed_model = compress_weights(model, mode=CompressWeightsMode.INT4_SYM, ratio=0.8, dataset=nncf_dataset) # model is openvino.Model object
 ```
 
+- Additionally, it is possible to use the model to generate a synthetic dataset based on the model output. For this purpose the `nncf.data.generate_text_data` method exits. This method returns the list of the strings generated by the model. Generation takes time and depends on various conditions like model size, requested dataset length or environment setup. Also, since the dataset was generated by model output, it does not guarantee significant accuracy improvement. This method is recommended to use only in case when any better dataset option is not available. Refer to the [example](https://github.com/openvinotoolkit/nncf/tree/develop/examples/llm_compression/openvino/tiny_llama_synthetic_data/) for the details of the usage.
```
Comment (MaximProshin) [inline docs/usage/post_training_compression/weights_compression/Usage.md:64]: I suggest to slightly rephrase it to:
Additionally, it is possible to generate a synthetic dataset by `nncf.data.generate_text_data` method to use it in the data-aware weight compression. The method takes a language model (e.g. from optimum.intel.openvino) and a tokenizer (e.g. from transformers) as input and returns the list of strings generated by the model. Note that the dataset generation takes time and depends on various conditions like the model size, requested dataset length or environment setup. Also, since the dataset is generated by the model output, it does not guarantee significant accuracy improvement after the compression. This method is recommended only in cases when a better dataset is not available. Refer to the [example](https://github.com/openvinotoolkit/nncf/tree/develop/examples/llm_compression/openvino/tiny_llama_synthetic_data/) for details of the usage.

Comment (KodiaqQ) [reply docs/usage/post_training_compression/weights_compression/Usage.md]: Done.

File: examples/llm_compression/openvino/tiny_llama_synthetic_data/main.py (hunk 2 above)
Comment (alexsu52) [inline examples/llm_compression/openvino/tiny_llama_synthetic_data/main.py:95]: Could you provide `fp_output_text` and `opt_output_text`?

Comment (KodiaqQ) [reply examples/llm_compression/openvino/tiny_llama_synthetic_data/main.py:95]: FP model output: What is Python? 

`Python is a high-level, interpreted, general-purpose programming language. It is designed to be easy to learn and use, and it has a large community of developers who contribute to its development. Python is widely used in a variety of fields, including data science, machine learning, web development, and more.`
`Python is a dynamic language, meaning that it dynamically types its code based on the data it receives. This means that Python can automatically detect the type of data being`

Compiling the model to CPU ...
Optimized model output: What is Python? 

`Python is a high-level, interpreted, interactive, and object-oriented programming language. It is widely used in various fields such as data science, artificial intelligence, web development, and machine learning. Python is a popular programming language because it is easy to learn, has a large community, and is widely used in various fields. Python is a dynamic language, which means that it dynamically types the data it receives. This means that Python can dynamically determine the data type of the data it`

Comment (alexsu52) [reply examples/llm_compression/openvino/tiny_llama_synthetic_data/main.py:95]: @MaximProshin, @KodiaqQ I would suggest not to output `fp_output_text`. It only interferes with assessing the correctness of the answer, and leads to an attempt to compare answers word by word.

Comment (KodiaqQ) [reply examples/llm_compression/openvino/tiny_llama_synthetic_data/main.py:90]: In this case, the example would be useless, for me.
But as you wish.
Done, removed.

**Review by alexsu52 — APPROVED** [review]
